backend/internal/server/apis/schedule/routes/upload/upload.go

```diff
@@ -0,0 +1,52 @@
+package upload
+
+import (
+	"fmt"
+	"github.com/gofiber/fiber/v2"
+	"github.com/jak103/powerplay/internal/server/apis"
+	"github.com/jak103/powerplay/internal/server/services/auth"
+	"github.com/jak103/powerplay/internal/utils/responder"
+	"mime/multipart"
+	"os"
+)
+
+func init() {
+	apis.RegisterHandler(fiber.MethodPost, "/schedule/upload", auth.Authenticated, handleUpload)
+}
+
+func handleUpload(c *fiber.Ctx) error {
+	file, err := c.FormFile("file")
+	if err != nil {
+		return responder.BadRequest(c, "File not found")
+	}
+
+	if !hasUploadsDirectory() {
+		if err = createUploadsDirectory(); err != nil {
+			return responder.InternalServerError(c, "Could not create uploads directory")
+		}
+	}
+
+	err = saveFile(c, file)
+	if err != nil {
+		return responder.InternalServerError(c, "Could not save file")
+	}
+
+	return responder.Ok(c, "File uploaded successfully")
+}
+
+func hasUploadsDirectory() bool {
+	_, err := os.Stat("./uploads")
```

Comment:
I don't think we would want the directory of files uploaded to be a relative path, because then it depends on where the pwd of the running program. I'm not sure where we would want to put these files.